http/action/src/test/java/io/knotx/databridge/http/action/HttpActionTest.java

```diff
@@ -546,13 +552,110 @@ void expectErrorWhenNotExistingPredicateProvided(VertxTestContext testContext, V
+    verifyFailingExecution(tested, clientRequest, createFragment(), error -> {
       assertTrue(error instanceof IllegalArgumentException);
     }, testContext);
   }
 
+  @Test
+  @DisplayName("expect success transition and action log should contain info level messages")
+  void actionLogShouldContainInfoLevelMessages(VertxTestContext testContext, Vertx vertx)
+      throws Throwable {
+    String endpointPath = "/api/info-action-log";
+
+    wireMockServer.stubFor(get(urlEqualTo(endpointPath))
+        .willReturn(aResponse().withBody(JSON_BODY)
+            .withHeader("Content-Type", APPLICATION_JSON)));
+
+    ClientRequest clientRequest = prepareClientRequest(MultiMap.caseInsensitiveMultiMap(),
+        MultiMap.caseInsensitiveMultiMap(), endpointPath);
+
+    EndpointOptions endpointOptions = new EndpointOptions()
+        .setPath(endpointPath)
+        .setDomain("localhost")
+        .setPort(wireMockServer.port())
+        .setAllowedRequestHeaderPatterns(Collections.singletonList(Pattern.compile(".*")));
+
+    HttpAction tested = new HttpAction(vertx,
+        new HttpActionOptions()
+            .setEndpointOptions(endpointOptions),
+        ACTION_ALIAS, ActionLogLevel.INFO);
+
+    verifyExecution(tested, clientRequest, createFragment(), fragmentResult -> {
+      assertNotNull(fragmentResult.getNodeLog().getMap().get("logs"));
+      JsonObject logs = fragmentResult.getNodeLog().getJsonObject("logs");
+      assertEquals(new JsonObject(JSON_BODY), logs.getJsonObject("_result"));
+      assertNotNull(logs.getString("rawBody"));
+      assertNotNull(logs.getJsonObject("_response"));
+    }, testContext);
+  }
+
+  @Test
+  @DisplayName("Expect success transition and action log should contain error level messages")
+  void actionLogShouldContainErrorLevelMessages(VertxTestContext testContext, Vertx vertx)
+      throws Throwable {
+    String endpointPath = "/api/error-action-log";
+
+    wireMockServer.stubFor(get(urlEqualTo(endpointPath))
+        .willReturn(aResponse().withBody(JSON_BODY)
+            .withHeader("Content-Type", APPLICATION_JSON)));
+
+    ClientRequest clientRequest = prepareClientRequest(MultiMap.caseInsensitiveMultiMap(),
+        MultiMap.caseInsensitiveMultiMap(), endpointPath);
+
+    EndpointOptions endpointOptions = new EndpointOptions()
+        .setPath(endpointPath)
+        .setDomain("localhost")
+        .setPort(wireMockServer.port())
+        .setAllowedRequestHeaderPatterns(Collections.singletonList(Pattern.compile(".*")));
+
+    HttpAction tested = new HttpAction(vertx,
+        new HttpActionOptions()
+            .setEndpointOptions(endpointOptions),
+        ACTION_ALIAS, ActionLogLevel.ERROR);
+
+    verifyExecution(tested, clientRequest, createFragment(), fragmentResult -> {
+      assertNotNull(fragmentResult.getNodeLog().getMap().get("logs"));
+      JsonObject logs = JsonObject.mapFrom(fragmentResult.getNodeLog().getMap().get("logs"));
+      assertEquals(0, logs.size());
```

Comment:
Please read test name and check what assertions are missing...

Reply:
fixed